designer/client/src/components/modals/ValidationLabels.tsx

```diff
@@ -54,3 +68,64 @@ export default function ValidationLabels(props: Props) {
         </LabelsContainer>
     );
 }
+
+type TemplateLabelProps = {
+    errorCode: string;
+    templateValues: TemplateValues;
+};
+
+const TemplatedType = styled.span({
+    fontWeight: "bold",
+    textDecoration: "underline",
+    "&:hover": {
+        color: "white",
+    },
+});
+
+function I18nTemplateLabel(props: TemplateLabelProps) {
+    const { errorCode, templateValues } = props;
+    const { i18n } = useTranslation();
+
+    const displayableTemplateValues = useMemo(() => {
+        return {
+            ...templateValues.stringValues,
+            ...Object.fromEntries(
+                Object.entries(templateValues.typingResultValues).map(([key, typingResult]) => {
+                    return [key, typingResult.display];
+                }),
+            ),
+        };
+    }, [templateValues]);
+
+    // we create separate spans only for TypingResult values
+    const templateComponents = useMemo(() => {
+        return Object.fromEntries(
+            Object.entries(templateValues.typingResultValues).map(([key, typingResult]) => {
+                const component = <TemplatedType title={typingResult.type} />;
+                return [key, component];
+            }),
+        );
+    }, [templateValues]);
+
+    const canRender =
+        i18n.exists(`nodeErrorCodes:${errorCode}.messageTemplate`) &&
+        validateTemplateInterpolation(
+            i18n.t(`nodeErrorCodes:${errorCode}.messageTemplate`),
+            Object.keys(displayableTemplateValues),
+            Object.keys(templateComponents),
+        );
+
+    return canRender ? (
+        <Trans
+            i18nKey={`nodeErrorCodes:${errorCode}.messageTemplate`}
+            values={displayableTemplateValues}
+            components={templateComponents}
+        ></Trans>
+    ) : null;
+}
+
+function validateTemplateInterpolation(template: string, valueNames: string[], componentNames: string[]): boolean {
```

Comment:
Do we need explicitly say that the return value is `boolean`? Ts is smart enough to calculate it bases on return value :)